### ISSUE TITLE
[Failer test] test_pe1_e1_i3.py

### ISSUE THREAD

**Issue by blitty-codes** [mit-ll/spacegym-kspdg#11]
# Testing failures

When executing the [pursuit-evasion (pe1_e1_i3)](https://github.com/mit-ll/spacegym-kspdg#verify-installation) the first test fails. A note, when selecting the mission, there are two "saves" but only one works (`0h 0m` on it). This may not be a problem, because maybe for testing there is one save and for the problem program there is another. Either way, it would be helpful to detail it.

![saves image](https://github.com/mit-ll/spacegym-kspdg/assets/54291870/2bded457-ee6a-4366-9c0c-acb7348079fe)

## Trazability

When loading the mission with `0h 0m`, and executing the test via `pytest tests/ksp_ingame_tests/test_pe1_e1_i3.py`, the following error is thrown.

```
tests\ksp_ingame_tests\test_pe1_e1_i3.py F.......                                                                [100%]

====================================================== FAILURES =======================================================
________________________________________ test_observation_dict_list_convert_0 _________________________________________

pe1_e1_i3_env = <kspdg.pe1.e1_envs.PE1_E1_I3_Env object at 0x0000023F723D3810>

    def test_observation_dict_list_convert_0(pe1_e1_i3_env):
        '''check that converting between lists and dict to not alter observation'''
        # ~~ ARRANGE ~~
        env = pe1_e1_i3_env

        # collect current observation as list
        obs_list = env.get_observation()

        # ~~ ACT ~~
        # convert obs to dict
        obs_dict = env.observation_list_to_dict(obs_list)

        # convert obs back to list
        obs_list_new = env.observation_dict_to_list(obs_dict)

        # ~~ ASSERT ~~
        # check that all observations are equal
>       assert np.allclose(obs_list, obs_list_new)

tests\ksp_ingame_tests\test_pe1_e1_i3.py:43:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tfm_env\Lib\site-packages\numpy\core\numeric.py:2241: in allclose
    res = all(isclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = [5.299999999999975, 6684.9755859375, 1199.992431640625, 707758.0091506173, -248150.0999515668, -0.0005483742130386832, ...]
b = [5.299999999999975, 6684.9755859375, 1199.992431640625, 707758.0091506173, -248150.0999515668, -0.0005483742130386832, ...]
rtol = 1e-05, atol = 1e-08, equal_nan = False

    @array_function_dispatch(_isclose_dispatcher)
    def isclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
        """
        Returns a boolean array where two arrays are element-wise equal within a
        tolerance.

        The tolerance values are positive, typically very small numbers.  The
        relative difference (`rtol` * abs(`b`)) and the absolute difference
        `atol` are added together to compare against the absolute difference
        between `a` and `b`.

        .. warning:: The default `atol` is not appropriate for comparing numbers
                     that are much smaller than one (see Notes).

        Parameters
        ----------
        a, b : array_like
            Input arrays to compare.
        rtol : float
            The relative tolerance parameter (see Notes).
        atol : float
            The absolute tolerance parameter (see Notes).
        equal_nan : bool
            Whether to compare NaN's as equal.  If True, NaN's in `a` will be
            considered equal to NaN's in `b` in the output array.

        Returns
        -------
        y : array_like
            Returns a boolean array of where `a` and `b` are equal within the
            given tolerance. If both `a` and `b` are scalars, returns a single
            boolean value.

        See Also
        --------
        allclose
        math.isclose

        Notes
        -----
        .. versionadded:: 1.7.0

        For finite values, isclose uses the following equation to test whether
        two floating point values are equivalent.

         absolute(`a` - `b`) <= (`atol` + `rtol` * absolute(`b`))

        Unlike the built-in `math.isclose`, the above equation is not symmetric
        in `a` and `b` -- it assumes `b` is the reference value -- so that
        `isclose(a, b)` might be different from `isclose(b, a)`. Furthermore,
        the default value of atol is not zero, and is used to determine what
        small values should be considered close to zero. The default value is
        appropriate for expected values of order unity: if the expected values
        are significantly smaller than one, it can result in false positives.
        `atol` should be carefully selected for the use case at hand. A zero value
        for `atol` will result in `False` if either `a` or `b` is zero.

        `isclose` is not defined for non-numeric data types.
        `bool` is considered a numeric data-type for this purpose.

        Examples
        --------
        >>> np.isclose([1e10,1e-7], [1.00001e10,1e-8])
        array([ True, False])
        >>> np.isclose([1e10,1e-8], [1.00001e10,1e-9])
        array([ True, True])
        >>> np.isclose([1e10,1e-8], [1.0001e10,1e-9])
        array([False,  True])
        >>> np.isclose([1.0, np.nan], [1.0, np.nan])
        array([ True, False])
        >>> np.isclose([1.0, np.nan], [1.0, np.nan], equal_nan=True)
        array([ True, True])
        >>> np.isclose([1e-8, 1e-7], [0.0, 0.0])
        array([ True, False])
        >>> np.isclose([1e-100, 1e-7], [0.0, 0.0], atol=0.0)
        array([False, False])
        >>> np.isclose([1e-10, 1e-10], [1e-20, 0.0])
        array([ True,  True])
        >>> np.isclose([1e-10, 1e-10], [1e-20, 0.999999e-10], atol=0.0)
        array([False,  True])
        """
        def within_tol(x, y, atol, rtol):
            with errstate(invalid='ignore'), _no_nep50_warning():
                return less_equal(abs(x-y), atol + rtol * abs(y))

        x = asanyarray(a)
        y = asanyarray(b)

        # Make sure y is an inexact type to avoid bad behavior on abs(MIN_INT).
        # This will cause casting of x later. Also, make sure to allow subclasses
        # (e.g., for numpy.ma).
        # NOTE: We explicitly allow timedelta, which used to work. This could
        #       possibly be deprecated. See also gh-18286.
        #       timedelta works if `atol` is an integer or also a timedelta.
        #       Although, the default tolerances are unlikely to be useful
        if y.dtype.kind != "m":
            dt = multiarray.result_type(y, 1.)
            y = asanyarray(y, dtype=dt)

>       xfin = isfinite(x)
E       TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

tfm_env\Lib\site-packages\numpy\core\numeric.py:2348: TypeError
------------------------------------------------ Captured stderr setup ------------------------------------------------
kspdg.base_envs - INFO - Connected to kRPC server
kspdg.base_envs - INFO - Changing active vehicle...
kspdg.base_envs - INFO - Activating Pursuer SAS, RCS and orienting to Evader...
------------------------------------------------- Captured log setup --------------------------------------------------
INFO     kspdg.base_envs:base_envs.py:90 Connected to kRPC server
INFO     kspdg.base_envs:pe1_base.py:199 Changing active vehicle...
INFO     kspdg.base_envs:pe1_base.py:205 Activating Pursuer SAS, RCS and orienting to Evader...
=============================================== short test summary info ===============================================
FAILED tests/ksp_ingame_tests/test_pe1_e1_i3.py::test_observation_dict_list_convert_0 - TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any su...
======================================= 1 failed, 7 passed in 107.01s (0:01:47) =======================================
```

**Comment by rallen10** [comment]
I can confirm that I am getting the same error. Debugging this now

**Comment by rallen10** [comment]
Fixed in release [v0.4.3](https://github.com/mit-ll/spacegym-kspdg/releases/tag/v0.4.3)

Note that the "two saves" of `pe1_i3` is an unrelated issue. Also note that there are `Missions/` files and `saves/missions/`. This appears to be a duplication in `Missions/`, not `saves/missions/`. 

Only one Mission directory for `pe1_i3` should be present on your machine, as is the case [in this repo](https://github.com/mit-ll/spacegym-kspdg/tree/main/ksp_files/Missions). I can't say exactly where you got the other, but it is likely an artifact of going into the "Making History" scenario editor and saving out another mission with the same name. In principle, you should be able to wipe out your `Missions` and `saves/missions` files in your KSP directory and re-install them fresh from this repo; but that might wipe out some locally-developed missions and/or saves you've created for other purposes.

**Comment by blitty-codes** [comment]
Thanks for the help!
